client/v3/concurrency/mutex.go

```diff
@@ -80,16 +79,31 @@ func (m *Mutex) Lock(ctx context.Context) error {
 		return nil
 	}
 	client := m.s.Client()
+
+	ch := make(chan bool, 1)
+	errorc := make(chan error, 1)
```

Comment:
What if instead the 'ch'  we would create 
nestedCtx, nestedCtxCancel := ctx.WithCancel()

and just call nestedCtxCancel() instead of ch <- true ?

Would that work ?